src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Resolve.java

```diff
@@ -458,6 +462,19 @@ public boolean isAccessible(Env<AttrContext> env, Type site, Symbol sym, boolean
             return isAccessible(env, site, checkInner) && notOverriddenIn(site, sym);
         }
     }
+
+    private boolean privateMemberInPermitsClauseIfAllowed(Env<AttrContext> env, Symbol sym) {
+        if (env.info.isPermitsClause && ((JCClassDecl) env.tree).sym.outermostClass() == sym.owner.outermostClass()) {
+            if (allowPrivateMembersInPermitsClause) {
+                return true;
+            } else {
+                log.error(env.tree.pos(), Errors.PrivateMembersInPermitsClause);
+                return false;
+            }
+        }
+        return false;
```

Comment:
shouldn't we return true here?

Reply:
No, method name is privateMemberInPermitsClause if allowed, and return is:
1. If condition in if is met, then return whether feature is enabled or not (value of field allowPrivateMembersInPermitsClause)
2. If condition is not met, then this is NOT privateMemberInPermitsClause, so return false

Reply:
what about:
```
private boolean privateMemberInPermitsClauseIfAllowed(Env<AttrContext> env, Symbol sym) {
    return allowPrivateMembersInPermitsClause &&
           env.info.isPermitsClause &&
           ((JCClassDecl) env.tree).sym.outermostClass() == sym.owner.outermostClass();
}
```